years/2019/01/solutions.ts

```diff
@@ -0,0 +1,16 @@
+export const part1 = (input: string[]): number => {
+  return input.reduce((acc: number, value: string) => (acc += calcuation(parseInt(value))), 0)
```

Comment:
It's common to avoid function param assignment:

<img width="817" alt="image" src="https://user-images.githubusercontent.com/24367010/228902454-844dc19d-e507-4e5c-a1bd-5ab969765bdb.png">

I also think it's not really doing anything here.

```suggestion
  return input.reduce((acc: number, value: string) => acc + calcuation(parseInt(value)), 0)
```

Reply:
> It's common to avoid function param assignment

Could you explain this one a little more for me please? I am not sure I fully understand this statement and would love to learn more.

Reply:
Is this basically, don't use `acc += 1`, just use `acc + 1`

Reply:
Basically yes. But also generally about avoiding re-assigning values to function parameters. `acc += 1` is the same as saying `acc = acc + 1`, so we're reusing the function param as a variable and _assigning_ the new value to it. Better to create a new variable inside the function, e.g.:

```tsx
const addOne = (num) => {
  return num + 1
}
```

or

```tsx
const addOne = (num) => {
  const newValue = num + 1

  return newValue
}
```

Here's the eslint rule about it with a little more explanation if you're interested: https://eslint.org/docs/latest/rules/no-param-reassign